File: 0494-target-sum/0494-target-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var findTargetSumWays = function(nums, target) {
    let res = 0
    const len = nums.length
    
    function helper(L, sum) {
        if (L === len) {
            if (sum === target) res += 1
            return
        }
        
        helper(L+1, sum + nums[L])
        helper(L+1, sum - nums[L])
        
    }
    
    helper(0, 0)
    
    return res
};

File: 0494-target-sum/0494-target-sum.cpp
class Solution {
public:
    int findTargetSumWays(vector<int>& nums, int S) {
        return find(0,nums,S);    
    }
    int find(int p, vector<int>& nums, int sum) {
        if(p==nums.size()) return sum==0;
        return find(p+1,nums,sum+nums[p])+find(p+1,nums,sum-nums[p]);
    }
};